controller_manager/src/ros2_control_node.cpp

```diff
@@ -33,36 +68,43 @@ int main(int argc, char ** argv)
 
   auto cm = std::make_shared<controller_manager::ControllerManager>(executor, manager_node_name);
 
-  // TODO(anyone): Due to issues with the MutliThreadedExecutor, this control loop does not rely on
-  // the executor (see issue #260).
-  // When the MutliThreadedExecutor issues are fixed (ros2/rclcpp#1168), this loop should be
-  // converted back to a timer.
+  RCLCPP_INFO(cm->get_logger(), "update rate is %d Hz", cm->get_update_rate());
+
   std::thread cm_thread(
     [cm]()
     {
-      RCLCPP_INFO(cm->get_logger(), "update rate is %d Hz", cm->get_update_rate());
+      if (has_realtime_kernel())
```

Comment:
So FIFO needs also real-time kernel? but can compile also on normal kernel?

Reply:
Yes, it compiles just fine because the code for configuring FIFO has made it into the mainline linux headers because they are part of the POSIX standard (just a part that mainline linux will error at runtime if you try to use).

Reply:
SCHED_DEADLINE however has not made it into mainline linux even though it is POSIX so you wouldn't be able to compile this code with normal linux.

How I'm using this is building it in CI into a docker where there is just normal linux, but then I pull that docker down onto my Debian machine that has a realtime kernel and it runs and uses these features.

Reply:
SCHED_FIFO and SCHED_RR are POSIX 1003.1, SCHED_DEADLINE is a Linux-only scheduler, and while I've head it has been ported to some other OSes, it's not part of any standard AFAIK. So you can't assume that SCHED_DEADLINE is available on other POSIXs/UNIXs like QNX or VxWorks either unfortunately.